src/layout/Address/AddressComponent.tsx

```diff
@@ -1,36 +1,39 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 
 import { LegacyTextField } from '@digdir/design-system-react';
 
 import { Label } from 'src/components/form/Label';
 import { FD } from 'src/features/formData/FormDataWrite';
 import { Lang } from 'src/features/language/Lang';
+import { ComponentValidations } from 'src/features/validation/ComponentValidations';
+import { useBindingValidationsForNode } from 'src/features/validation/selectors/bindingValidationsForNode';
+import { useComponentValidationsForNode } from 'src/features/validation/selectors/componentValidationsForNode';
+import { hasValidationErrors } from 'src/features/validation/utils';
+import { usePostPlaceQuery } from 'src/hooks/queries/usePostPlaceQuery';
 import classes from 'src/layout/Address/AddressComponent.module.css';
 import type { PropsFromGenericComponent } from 'src/layout';
-import type { IDataModelBindingsForAddressInternal } from 'src/layout/Address/config.generated';
 
 export type IAddressComponentProps = PropsFromGenericComponent<'AddressComponent'>;
 
-type AddressKey = keyof IDataModelBindingsForAddressInternal;
-
 export function AddressComponent({ node }: IAddressComponentProps) {
   const { id, required, readOnly, labelSettings, simplified, saveWhileTyping } = node.item;
 
+  const bindingValidations = useBindingValidationsForNode(node);
+  const componentValidations = useComponentValidationsForNode(node);
+
   const bindings = ('dataModelBindings' in node.item && node.item.dataModelBindings) || {};
   const saveData = FD.useSetForBindings(bindings, saveWhileTyping);
   const debounce = FD.useDebounceImmediately();
   const { address, careOf, postPlace, zipCode, houseNumber } = FD.usePickFreshStrings(bindings);
 
-  const onSaveField = React.useCallback(
-    (key: AddressKey, value: any) => {
-      saveData(key, value);
-      if (key === 'zipCode' && !value) {
-        // if we are removing a zip code, also remove post place from form data
-        saveData('postPlace', '');
-      }
-    },
-    [saveData],
-  );
+  const postPlaceQueryData = usePostPlaceQuery(zipCode, !hasValidationErrors(bindingValidations?.zipCode));
+  useEffect(() => {
+    if (postPlaceQueryData != null && postPlaceQueryData != postPlace) {
+      saveData('postPlace', postPlaceQueryData);
+    }
+    // TODO(useDelayedSavedState): This hook could disappear, but for now, a problem is that the function references are very volatile and will cause many rerenders
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Is this still relevant?

Reply:
I will check it out, have not looked into it since the experiment was merged 👍